### DISCUSSION TITLE
Noble/6.1 Kernel stability

### DISCUSSION THREAD

**Discussion by Xeue** [Joshua-Riek/ubuntu-rockchip#686] [General]
Hey, awsome work with 6.1 and Noble, this is going to be super useful! Getting gstreamer up to version 24 is a huge win for me as well!
I'm having a few issue, just wanted to know if they are likly to do with the WIP status of it all or if there is potentially more at play.

I revisted trying to install Blackmagic Desktop Video and it hits and issue during the apt install when it gets to the stage of loading the DKMS/Kernel driver and the entire system just freezes. No kernel logs or anything... It's likely it's a blackmagic driver issue, but I wanted to know if there was anything DKMS related that cause be a cause?

I also find that when running some programs (mostly on a little NodeJS app I've been building) that it randomly quits and states 'Bus error (core dumped)' or various different I/O errors. After this has happened most programs can't be run with I/O errors, even tings like ls, nano, cat etc. I'm at a bit of a loss on where to even beging trouble shooting. I've tried the latest image from actions on a couple of SD cards to the same results.

Once again, fantastic work, it's hugely appreciated!

**Comment by Xeue** [comment]
PS. the wifi issues with the AX210 also appear to have been resolved now!

**Reply by Joshua-Riek** [reply]
I would not use any noble images right now due to CVE-2024-3094, the official Ubuntu 24.04 Beta and Release dates have been pushed back because of this. They are currently rebuilding every single package so I can not do anything until then.

I would need some logs to figure out whats going on with your two issues, but i do know DKMS will work in general, I tested it with ZFS. It's highly likely that some of this may be addressed by upstream by the official release date.

**Reply by Xeue** [reply]
That's pretty much what I expected! I'm only using it in an isolated environment for some dev at the moment, fully aware of the xz issue!

I'll see what logs I can find, but I'll wait for upstream before digging too much!

Thanks again

**Comment by Xeue** [comment]
Bit of a random one, how do you apply device overlays now it's extlinux/u-boot? I think trying to enable the i2c port on my box was causing some of my issues...

**Reply by Joshua-Riek** [reply]
Ive not documented it yet, but in the latest builds -

1. Open the file /etc/default/u-boot
2. Modify U_BOOT_FDT_OVERLAYS with the overlay
```
## /etc/default/u-boot - configuration file for u-boot-update(8)

#U_BOOT_UPDATE="true"

#U_BOOT_ALTERNATIVES="default recovery"
#U_BOOT_DEFAULT="l0"
#U_BOOT_PROMPT="1"
#U_BOOT_ENTRIES="all"
#U_BOOT_MENU_LABEL="Debian GNU/Linux"
#U_BOOT_PARAMETERS="ro earlycon"
#U_BOOT_ROOT=""
#U_BOOT_TIMEOUT="50"
#U_BOOT_FDT=""
#U_BOOT_FDT_DIR="/lib/firmware/"
U_BOOT_FDT_OVERLAYS="device-tree/rockchip/overlay/rk3588-pwm14-m1.dtbo"
#U_BOOT_FDT_OVERLAYS_DIR="/lib/firmware/"
#U_BOOT_SYNC_DTBS="false"
```
3. Save and enter the command `sudo u-boot-update`

**Reply by Joshua-Riek** [reply]
You can then verify the overlay is found:
```
ubuntu@ubuntu:~$ cat /boot/extlinux/extlinux.conf 
## /boot/extlinux/extlinux.conf
##
## IMPORTANT WARNING
##
## The configuration of this file is generated automatically.
## Do not edit this file manually, use: u-boot-update

default l0
menu title U-Boot menu
prompt 1
timeout 10


label l0
	menu label Ubuntu Noble Numbat (development branch) 6.1.0-1006-rockchip
	linux /boot/vmlinuz-6.1.0-1006-rockchip
	initrd /boot/initrd.img-6.1.0-1006-rockchip
	fdtdir /lib/firmware/6.1.0-1006-rockchip/device-tree/
	fdtoverlays  /lib/firmware/6.1.0-1006-rockchip/device-tree/rockchip/overlay/rk3588-pwm14-m1.dtbo
	append root=UUID=17c66edb-0f6d-4ab9-b080-0a98422994c7 rootwait rw console=ttyS2,1500000 console=tty1 cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory

label l0r
	menu label Ubuntu Noble Numbat (development branch) 6.1.0-1006-rockchip (rescue target)
	linux /boot/vmlinuz-6.1.0-1006-rockchip
	initrd /boot/initrd.img-6.1.0-1006-rockchip
	fdtdir /lib/firmware/6.1.0-1006-rockchip/device-tree/
	append root=UUID=17c66edb-0f6d-4ab9-b080-0a98422994c7 rootwait rw console=ttyS2,1500000 console=tty1 cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory single
```

**Comment by Xeue** [comment]
Hey, I'm being dumb again, how can I set the vmalloc parameter? Is there a way to do that with u-boot or do I have to compile it into the kernel?
Ps. do you know what the default would be? 128M? Or higher?

**Reply by Joshua-Riek** [reply]
You need to modify the kernel command line:

https://github.com/Joshua-Riek/ubuntu-rockchip/wiki/Ubuntu-24.04#changing-the-linux-boot-parameters
https://askubuntu.com/questions/1125306/how-to-set-virtual-memory-allocation-vmalloc-on-linux-to-avoid-kernel-driver-c

**Reply by Xeue** [reply]
I was being dumb... Didn't see the kernel command line section on that page!
